src/Symfony/Bridge/Twig/Extension/RoutingExtension.php

```diff
@@ -26,10 +27,12 @@
 class RoutingExtension extends AbstractExtension
 {
     private $generator;
+    private $router;
 
-    public function __construct(UrlGeneratorInterface $generator)
+    public function __construct(UrlGeneratorInterface $generator, RouterInterface $router)
```

Comment:
adding a new mandatory arg is a BC break: you need to make it optional and deal with this possibility in the code

Reply:
Thus, this makes it impossible to use this extension in a context not using the full RouterInterface, but using only the UrlGeneratorInterface and MatcherInterface interfaces

Reply:
correct, and actually having an `UrlGeneratorInterface` is enough to implement the logic!